### ISSUE TITLE
Add message in JUnit report in case of pass assertion

### ISSUE THREAD

**Issue by anandmeher** [junit-team/junit5#703]
- ( ) **Feature request.** To make the report more readable and proper, please add the success message if assertion passed.

like Assertions.assertEquals( "John", "John1" ,"Passed - Name is verified", "Failed - Name is not same" 

## Deliverables

- [ ] ...

**Comment by sormuras** [comment]
Does the reporting feature described in http://junit.org/junit5/docs/current/user-guide/#writing-tests-dependency-injection as _The TestReporter can be used to publish additional data about the current test run._ help here?
```
@Test
void verifyName(TestReporter testReporter) {
    Assertions.assertEquals( "John", "John1", "Failed - Name is not same"):
    testReporter.publishEntry("Passed", "Name is verified");
}
```
I don't think having success messages has benefits beyond what you can do with naming your test methods as clearly as you can.

**Comment by nicolaiparlog** [comment]
@anandmeher By and large an assertion is just an `if` that checks a property and throws an exception if it is not fulfilled - there is no deeper interaction with the testing framework. This is what makes it possible to exchange assertion libraries. Making passed assertions log messages through the framework would be a fundamental change.

On top of that, the benefits are questionable. Assertions should work on behalf of the method they are a part of. Hence a passing method should suffice.

**Comment by nicolaiparlog** [comment]
I would close this with _wontfix_, _works-as-designed_.

**Comment by gaganis** [comment]
It might be that the _One Assertion per Test_ rule of thumb is relevant to this point. If we have only one assertion per Test, it's name would very closely reflect the actual assertion that is being checked.

If you have a test with many assertions it would be more difficult to tie the test's name with all the assertions, and seeing that name in  a run report would not be as descriptive. In a case like this a listing of all the assertions would be helpful. 

I have been in situation where having many assertions per test made us also feel that we needed a more detailed report of the passing assertions, and in our case we made things clearer by breaking our tests to smaller ones. We did not reach one per test but we reduced the number per test a lot. And in our case his helped us a lot to reason about what exactly our units were actually testing.

@anandmeher If it applies to your case you might want to also consider trying to apply this rule.
